cms/static/js/views/list.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * A generic list view class.
+ *
+ * Expects the following properties to be overriden:
+ * - emptyTemplateName (string): Name of an underscore template to
+ *   render when the collection is empty.
+ * - createItemView (function): Create and return an item view for a
+ *   model in the collection.
+ * - newModelOptions (object): Options to pass to models which are
+ *   added to the collection.
+ * - itemCategoryDisplayName (string): Display name for the category
+ *   of items this list contains.  For example, 'Group Configuration'.
+ *   Note that it must be translated.
+ */
+define([
+    'js/views/baseview'
+], function(BaseView) {
+    'use strict';
+    var ListView = BaseView.extend({
+        events: {
+            'click .action-add': 'onAddItem',
+            'click .new-button': 'onAddItem'
+        },
+
+        initialize: function() {
+            this.emptyTemplate = this.loadTemplate(this.emptyTemplateName);
+            this.listenTo(this.collection, 'add', this.addNewItemView);
+            this.listenTo(this.collection, 'remove', this.onRemoveItem);
+            this.template = _.template([
```

Comment:
Just for the record: I suggested inlining the add template into the parent template, not realizing that there wasn't one. I don't like having the template inlined in the code.

I really would prefer to have a parameterized template for the list view itself, and to have it contain the empty message, the add buttons etc. I guess I haven't thought through enough about how the two concrete subclasses are working currently, so I'm not suggesting that this is a refactoring to do now.

I would rename this as ```this.addButtonTemplate``` to reduce confusion, as every other view has a ```template``` field which renders itself.